libs/model/src/services/tokenBalanceCache/providers/get_spl_balances.ts

```diff
@@ -0,0 +1,102 @@
+import { logger } from '@hicommonwealth/logging';
+import { getAccount, getAssociatedTokenAddressSync } from '@solana/spl-token';
+import { Connection, PublicKey } from '@solana/web3.js';
+import { ChainNodeInstance } from 'model/src/models/chain_node';
+import { fileURLToPath } from 'url';
+import { Balances, GetSPLBalancesOptions } from '../types';
+import { failingChainNodeError } from '../util';
+
+const __filename = fileURLToPath(import.meta.url);
+const log = logger(__filename);
+
+export async function __get_spl_balances(
+  chainNode: ChainNodeInstance,
+  options: GetSPLBalancesOptions,
+): Promise<Balances> {
+  const rpcEndpoint = chainNode.private_url || chainNode.url;
+  if (options.addresses.length === 1) {
+    const balances: Balances = {};
+    balances[options.addresses[0]] = await getSingleSPLBalance(
+      new Connection(rpcEndpoint),
+      options.mintAddress,
+      options.addresses[0],
+    );
+    return balances;
+  }
+
+  return (
+    await getBulkSPLBalances(
+      rpcEndpoint,
+      options.mintAddress,
+      options.addresses,
+      options.batchSize,
+    )
+  ).balances;
+}
+
+async function getBulkSPLBalances(
+  rpcEndpoint: string,
+  mintAddress: string,
+  addresses: string[],
+  batchSize = 100,
+): Promise<{ balances: Balances; failedAddresses: string[] }> {
+  const connection = new Connection(rpcEndpoint);
+  let batchRequestPromises = [];
+  const idAddressMap: Balances = {};
+  const balances: Balances = {};
+  let failedAddresses: string[] = [];
+  let id = 0;
+  for (
+    let startIndex = 0;
+    startIndex < addresses.length;
+    startIndex += batchSize
+  ) {
+    const endIndex = Math.min(startIndex + batchSize, addresses.length);
+    const batchAddresses = addresses.slice(startIndex, endIndex);
+    for (const address of batchAddresses) {
+      batchRequestPromises.push(
+        getSingleSPLBalance(connection, mintAddress, address),
+      );
+      idAddressMap[id] = address;
+      ++id;
+    }
+    const responses = await Promise.allSettled(batchRequestPromises);
```

Comment:
If we can use `getMultipleAccounts` that will be better for this as well since `allSettled` won't throw for a rejected promise but our error handling is set up so that errors propagate up to the global `getBalances` method which has the proper log message + context.

Reply:
Lmk your thoughts given the comment above re: getMultipleAccounts limitations.